src/TLuaInterpreter.cpp

```diff
@@ -1314,27 +1270,15 @@ int TLuaInterpreter::selectCaptureGroup(lua_State* L)
 int TLuaInterpreter::getLines(lua_State* L)
 {
     int n = lua_gettop(L);
-    int s = 0;
+    int s = 1;
     QString windowName;
     if (n > 2) {
-        if (!lua_isstring(L, ++s)) {
-            lua_pushfstring(L, "getLines: bad argument #%d type (mini console, user window or buffer name as string expected {may be omitted for the \"main\" console}, got %s!)", s, luaL_typename(L, s));
-            return lua_error(L);
-        }
-        windowName = lua_tostring(L, s);
-    }
-
-    if (!lua_isnumber(L, ++s)) {
-        lua_pushfstring(L, "getLines: bad argument #%d type (start line as number expected, got %s!)", s, luaL_typename(L, s));
-        return lua_error(L);
-    }
-    int lineFrom = lua_tointeger(L, s);
-
-    if (!lua_isnumber(L, ++s)) {
-        lua_pushfstring(L, "getLines: bad argument #%d type (end line as number expected, got %s!)", s, luaL_typename(L, s));
-        return lua_error(L);
+        windowName = getVerifiedString(L, __func__, 1, "mini console, user window or buffer name {may be omitted for the \"main\" console}", true);
+        s++;
     }
-    int lineTo = lua_tointeger(L, s);
+    int lineFrom = getVerifiedInt(L, __func__, s, "start line");
+    s++;
```

Comment:
I'd suggest removing this and incorporating it as a pre-increment in the next line...

Reply:
There were a WHOLE LOT of s++ and ++s shenanigans which I for one found quite irritating, so I try and untangle them where possible. I find explicit and longer sometimes more readable than extreme incorporation. Then again, maybe I sometimes over-simplify. May need to review later.

Reply:
I have now incorporated all stand-alone s++ and ++s in some other adjacent command.
Would like more feedback if this becomes more readable this way or actually not.